pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1350,7 +1350,9 @@ func (m *kubeGenericRuntimeManager) SyncPod(ctx context.Context, pod *v1.Pod, po
 		}
 
 		// NOTE (aramase) podIPs are populated for single stack and dual stack clusters. Send only podIPs.
-		if msg, err := m.startContainer(ctx, podSandboxID, podSandboxConfig, spec, pod, podStatus, pullSecrets, podIP, podIPs, imageVolumes); err != nil {
+		msg, err = m.startContainer(ctx, podSandboxID, podSandboxConfig, spec, pod, podStatus, pullSecrets, podIP, podIPs, imageVolumes)
+		incrementImageVolumeMetrics(err, spec.container, imageVolumes)
```

Comment:
How do we identify the error type? The description of metrics tracks the number of failed image volume mounts. We can not say that all the errors are due to the image volume mount.

Reply:
Yes, because we did not define any specific error. What we can state is that there was definitely no mount due to an error. At least that's how I interpreted it.

Reply:
Can we define a CRI error for tracking image mount failures and propagate that? I think without that we can't clearly know which part of the pod startup is failing and how image mount is contributing to that.

Reply:
Yes, updated the implementation to work with a cri-api error. :+1: